bootstrapper/internal/kubernetes/k8sapi/kubectl/kubectl.go

```diff
@@ -123,30 +109,71 @@ func (k *Kubectl) AddTolerationsToDeployment(ctx context.Context, tolerations []
 //
 // [K8s selectors]: https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/
 func (k *Kubectl) AddNodeSelectorsToDeployment(ctx context.Context, selectors map[string]string, name string, namespace string) error {
-	client, err := k.clientGenerator.NewClient(k.kubeconfig)
-	if err != nil {
-		return err
-	}
+	deployments := k.AppsV1().Deployments(namespace)
+
+	// retry resource update if an error occurs
+	err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		result, err := deployments.Get(ctx, name, metav1.GetOptions{})
+		if err != nil {
+			return fmt.Errorf("failed to get Deployment to add node selector: %v", err)
+		}
+
+		for k, v := range selectors {
+			result.Spec.Template.Spec.NodeSelector[k] = v
+		}
 
-	if err = client.AddNodeSelectorsToDeployment(ctx, selectors, name, namespace); err != nil {
+		if _, err = deployments.Update(ctx, result, metav1.UpdateOptions{}); err != nil {
+			return err
+		}
+		return nil
+	})
+	if err != nil {
 		return err
 	}
-
 	return nil
 }
 
 // WaitForCRDs waits for a list of CRDs to be established.
 func (k *Kubectl) WaitForCRDs(ctx context.Context, crds []string) error {
-	client, err := k.clientGenerator.NewClient(k.kubeconfig)
-	if err != nil {
-		return err
-	}
-
 	for _, crd := range crds {
-		err = client.WaitForCRD(ctx, crd)
+		err := k.WaitForCRD(ctx, crd)
 		if err != nil {
 			return err
 		}
 	}
 	return nil
 }
+
+// WaitForCRD waits for the given CRD to be established.
+func (k *Kubectl) WaitForCRD(ctx context.Context, crd string) error {
```

Comment:
```suggestion
// waitForCRD waits for the given CRD to be established.
func (k *Kubectl) waitForCRD(ctx context.Context, crd string) error {
```
Does not need to be exposed